File: Cal State San Bernardino (CSUSB)/CSE375 - Requirements of Analysis & Design/CSE375 Website/cse.csusb.edu/dick/cs375/Die.cpp
/* Declares a class in Page 10 of Larman */
#ifndef DIECPP
#define DIECPP
#include <cstdlib>
#include <ctime>
using namespace std;

class Die
{ protected:  //future proofing to demo polymorphism
	int faceValue;
  public:
	virtual void roll()
	{
		faceValue = 1+(6.0*rand())/RAND_MAX;
	}
	virtual int getFaceValue()
	{
		return faceValue;
	}
	Die () { roll(); };
private:
	static const int seeded;
	static int seed(){ srand(time(NULL)); return 0; }
};
const int Die::seeded=seed();
#endif

File: Cal State San Bernardino (CSUSB)/CSE375 - Requirements of Analysis & Design/CSE375 Website/cse.csusb.edu/dick/cs375/Die.java
/** Declares a class in Page 10 of Larman */
import java.util.Random;
public class Die
{ protected int faceValue;
  protected Random rand;
  public void roll()
	{
		faceValue = 1+rand.nextInt(6);
	}
  public int getFaceValue()
	{
		return faceValue;
	}
  public Die () {rand=new Random(); roll(); };


   public static void main(String argv[])
   /** Unit test of a Die class */
   {
	Die die=new Die();

	int SAMPLE =1000;
	for(int i=0; i<SAMPLE; i++)
	{
		die . roll();
		int fv = die . getFaceValue();
		System.out.println(fv);
	}
	

   }//main
}//Die
